corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -17,11 +17,69 @@ hqDefine("geospatial/js/case_grouping_map",[
         self.coordinates = coordiantes;
         self.caseLink = caseLink;
 
+        // TODO: Group ID needs to be set
+        self.groupId = null;
+
         return self;
     }
 
+    function exportModel() {
+        var self = {};
+
+        self.casesToExport = ko.observableArray([]);
+
+        self.handleExportCSV = function () {
+            const casesToExport = _.map(self.casesToExport(), function (caseItem) {
+                const coordinates = (caseItem.coordinates) ? `${caseItem.coordinates.lng} ${caseItem.coordinates.lat}` : "";
+                return {
+                    'groupId': caseItem.groupId,
+                    'caseId': caseItem.caseId,
+                    'coordinates': coordinates,
+                };
+            });
+            exportToCsv(casesToExport);
+        };
+
+        // If list of propertiesToInclude is not given, will export all properties of objects in itemsArr
+        function exportToCsv(itemsArr, includeHeaders = true) {
+            if (!itemsArr.length) {
+                return;
+            }
+
+            let csvStr = "";
+            if (includeHeaders) {
+                csvStr = Object.keys(itemsArr[0]).join(",");
+                csvStr += "\n";
+            }
+
+            _.forEach(itemsArr, function (itemRow) {
+                csvStr += Object.keys(itemRow).map(key => itemRow[key]).join(",");
+                csvStr += "\n";
+            });
+
+            // Download CSV file
+            const hiddenElement = document.createElement('a');
+            hiddenElement.href = 'data:text/csv;charset=utf-8,' + encodeURI(csvStr);
+            hiddenElement.target = '_blank';
+            hiddenElement.download = `cases_by_group (created on ${getTodayDate()}).csv`;
+            hiddenElement.click();
+            hiddenElement.remove();
+        }
+
+        return self;
+    }
+
+    function getTodayDate() {
+        const todayDate = new Date();
+        const day = String(todayDate.getDate()).padStart(2, '0');
+        const month = String(todayDate.getMonth() + 1).padStart(2, '0');  // January is 0
+        const year = todayDate.getFullYear();
+        return `${year}-${month}-${day}`;
```

Comment:
oh no, we can't tell the format we need and simply get that?

Reply:
After looking around a bit, I do see there is a useful `toLocaleDateString()` function we can rather use to put it in the user's local format. I'll update it with this implementation.

Reply:
[1058ecc](https://github.com/dimagi/commcare-hq/pull/33610/commits/1058eccd309043e5203a4ce759fd9fe300c3df8d)

Reply:
Amazing. It would be crazy if there was no function for this.